src/Microsoft.DotNet.Darc/DarcLib/IGitRepoCloner.cs

```diff
@@ -27,5 +29,12 @@ public interface IGitRepoCloner
     /// <param name="repoUri">Repository uri to clone</param>
     /// <param name="targetDirectory">Target directory to clone to</param>
     /// <param name="gitDirectory">Location for the .git directory, or null for default</param>
-    public void Clone(string repoUri, string targetDirectory, string? gitDirectory);
+    public Task CloneAsync(string repoUri, string targetDirectory, string? gitDirectory);
+}
+
+public enum CheckoutType
+{
+    CheckoutWithoutSubmodules,
+    CheckoutWithSubmodules,
+    NoCheckout,
```

Comment:
Just a nit - for enums, the semantic equivalent of "None" should go as the first (default) element.